static/rtl-view.js

```diff
@@ -0,0 +1,205 @@
+    var options = require('options');
+    var _ = require('underscore');
+
+    require('asm-mode');
+    require('selectize');
+
+    function Rtl(hub, container, state) {
+        this.container = container;
+        this.eventHub = hub.createEventHub();
+        this.domRoot = container.getElement();
+        this.domRoot.html($('#rtl').html());
+        this.compilers = {};
+        this._currentDecorations = [];
+        this.rtlEditor = monaco.editor.create(this.domRoot.find(".monaco-placeholder")[0], {
+            value: "",
+            scrollBeyondLastLine: false,
+//            language: 'cppp', //we only support cpp for now
+            readOnly: true,
+            glyphMargin: true,
+            quickSuggestions: false,
+            fixedOverflowWidgets: true,
+            minimap: {
+                maxColumn: 80
+            },
+            lineNumbersMinChars: 3
+        });
+
+	var selectize = this.domRoot.find(".rtl-pass-picker").selectize({
+	    sortField: 'name',
+            valueField: 'name',
+            labelField: 'name',
+            searchField: ['name'],
+            options: [],
+            items: [],
+            render: {
+                option: function (item, escape) {
+                    return '<div><span class="options">' + escape(item.name) + '</span></div>';
+                }
+            },
+	    onChange : _.bind(function (value) {
+		this.onPassSelect(value);
+	    }, this),
+	});
+
+	this.selectize  = selectize[0].selectize;
+
+        this._compilerid = state.id;
+        this._compilerName = state.compilerName;
+        this._editorid = state.editorid;
+        this.fontScale = new FontScale(this.domRoot, state, this.rtlEditor);
+        this.fontScale.on('change', _.bind(this.updateState, this));
+
+        this.eventHub.on('compileResult', this.onCompileResult, this);
+        this.eventHub.on('compiler', this.onCompiler, this);
+        this.eventHub.on('compilerClose', this.onCompilerClose, this);
+        this.eventHub.on('editorChange', this.onEditorChange, this);
+        this.eventHub.on('settingsChange', this.onSettingsChange, this);
+
+        this.eventHub.emit('rtlViewOpened', this._compilerid);
+        this.eventHub.emit('requestSettings');
+        this.container.on('destroy', function () {
+            this.eventHub.emit("rtlViewClosed", this._compilerid);
+            this.eventHub.unsubscribe();
+            this.rtlEditor.dispose();
+        }, this);
+
+        container.on('resize', this.resize, this);
+        container.on('shown', this.resize, this);
+        if (state && state.rtlOutput) {
+            this.showRtlResults(state.rtlOutput);
+        }
+        this.setTitle();
+    }
+
+    Rtl.prototype.onPassSelect = function (id) {
+        // this.eventHub.emit('resendCompilation', id);
+	this.eventHub.emit('rtlPassSelected', this._compilerid,
+			   {
+			       selectedPass: id
+			   });
+        // this.updateCompilerNames();
+        // this.updateState();
```

Comment:
You prob want to store an state so this panel gets sync'ed on url sharing/page refresh. It is super easy. Try to store as little as possible. Just enough to be able to recreate the view when the state was saved (Even if that means recompiling)